src/test/java/com/savvato/tribeapp/controllers/ConnectAPITest.java

```diff
@@ -229,4 +229,33 @@ public void saveCosign() throws Exception {
 
     }
 
+    @Test
+    public void deleteCosign() throws Exception {
+        when(userPrincipalService.getUserPrincipalByEmail(Mockito.anyString()))
+                .thenReturn(new UserPrincipal(user));
+        String auth = AuthServiceImpl.generateAccessToken(user);
+
+        Long userIdIssuing = 1L;
+        Long userIdReceiving = 1L;
+        Long phraseId = 1L;
+
+        CosignRequest cosignRequest = new CosignRequest();
+        cosignRequest.userIdIssuing = userIdIssuing;
+        cosignRequest.userIdReceiving = userIdReceiving;
+        cosignRequest.phraseId = phraseId;
+
+        doNothing().when(cosignService).deleteCosign(anyLong(), anyLong(), anyLong());
+
+        this.mockMvc
```

Comment:
Add a test for what happens when an exception is thrown from the service.

Reply:
I added the test and updated the connect controller and the cosign service. I can further discuss in our meeting tomorrow to see if this is how you want it handled.